Azure/azure-cli
# {Core} Improve autocomplete performance by preventing loading all modules

**Description**

Close #24762, related issue #9157, related pr #13294
If the command is not complete, for example: `az account sho`, press TAB, cli loads all modules, which takes 1s. It gets worse if user installs many extensions. This make autocomplete slow.
This PR prevents module loading in autocomplete mode.

If command is `az acc`:
1. module begin with `acc` will be returned in `CommandIndex.get()` .
2. `raw_cmd` is 'acc'. Trim last word, it becomes `''`, return `self.command_table`.
```
cli.azure.cli.core: In autocomplete mode, load command begin with: 'acc'
cli.azure.cli.core: Modules found from index for 'acc': ['azure.cli.command_modules.profile', 'azure.cli.command_modules.resource']

cli.azure.cli.core: Could not find a match in the command or command group table for 'acc'
cli.azure.cli.core: In autocomplete mode, try to match trimmed raw cmd: ''
cli.azure.cli.core: Trimmed raw cmd is empty, return command table.
```

If command is `az account sh`:
1. `account` module is returned in `CommandIndex.get()` .
2. `raw_cmd` is `account sh`. Trim last word, it becomes `account` and is in `self.command_group_table`, return `self.command_table`.
```
cli.azure.cli.core: Modules found from index for 'account': ['azure.cli.command_modules.profile', 'azure.cli.command_modules.resource']

cli.azure.cli.core: Could not find a match in the command or command group table for 'account sh'
cli.azure.cli.core: In autocomplete mode, try to match trimmed raw cmd: 'account'
cli.azure.cli.core: Found a match in the command group table for trimmed raw cmd: 'account'.
```

**Testing Guide**
`az acco` and `az account sh` should be faster.
```
$env:_ARGCOMPLETE=1;
Measure-Command {az account sh}
TotalMilliseconds : 567.0021

Remove-Item Env:\_ARGCOMPLETE;
Measure-Command {az account sh}
TotalMilliseconds : 2935.8341
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.
